core/src/main/java/io/github/fourlastor/game/demo/state/unit/effect/ChargeBeamEffect.kt

```diff
@@ -0,0 +1,50 @@
+package io.github.fourlastor.game.demo.state.unit.effect
+
+import com.badlogic.gdx.assets.AssetManager
+import com.badlogic.gdx.graphics.Color
+import com.badlogic.gdx.scenes.scene2d.Action
+import com.badlogic.gdx.scenes.scene2d.actions.Actions
+import com.badlogic.gdx.scenes.scene2d.ui.Label
+import dagger.assisted.Assisted
+import dagger.assisted.AssistedFactory
+import dagger.assisted.AssistedInject
+import io.github.fourlastor.game.demo.state.unit.Mon
+import io.github.fourlastor.game.demo.state.unit.effect.Effect.Triggered
+
+class ChargeBeamEffect @AssistedInject constructor(
+    @Assisted mon: Mon,
+    assetManager: AssetManager
+) : Triggered {
```

Comment:
I prefer to write this as `Effect.Triggered` as it's a bit simpler to see this is an effect that way